src/windows/common/WslClient.cpp

```diff
@@ -1539,6 +1541,159 @@ int RunDebugShell()
+    parser.AddArgument(Integer(settings.MemoryMb), L"--memory");
+    parser.AddArgument(Integer(settings.CpuCount), L"--cpu");
+    parser.AddArgument(help, L"--help");
+
+    parser.Parse();
+
+    if (help)
+    {
+        const auto usage = std::format(
+            LR"({} --wsla [--vhd </path/to/vhd>] [--shell </path/to/shell>] [--memory <memory-mb>] [--cpu <cpus>] [--dns-tunneling] [--help])",
+            WSL_BINARY_NAME);
+
+        wprintf(L"%ls\n", usage.c_str());
+        return 1;
+    }
+
+    wil::com_ptr<IWSLAUserSession> userSession;
+    THROW_IF_FAILED(CoCreateInstance(__uuidof(WSLAUserSession), nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(&userSession)));
+    wsl::windows::common::security::ConfigureForCOMImpersonation(userSession.get());
+
+    wil::com_ptr<IWSLAVirtualMachine> virtualMachine;
+    THROW_IF_FAILED(userSession->CreateVirtualMachine(&settings, &virtualMachine));
+    wsl::windows::common::security::ConfigureForCOMImpersonation(userSession.get());
+
+    wil::unique_cotaskmem_ansistring diskDevice;
+    ULONG Lun{};
+    THROW_IF_FAILED(virtualMachine->AttachDisk(vhd.c_str(), true, &diskDevice, &Lun));
+
+    THROW_IF_FAILED(virtualMachine->Mount(diskDevice.get(), "/mnt", "ext4", "ro", WslMountFlagsChroot | WslMountFlagsWriteableOverlayFs));
+    THROW_IF_FAILED(virtualMachine->Mount(nullptr, "/dev", "devtmpfs", "", 0));
+    THROW_IF_FAILED(virtualMachine->Mount(nullptr, "/sys", "sysfs", "", 0));
+    THROW_IF_FAILED(virtualMachine->Mount(nullptr, "/proc", "proc", "", 0));
+    THROW_IF_FAILED(virtualMachine->Mount(nullptr, "/dev/pts", "devpts", "noatime,nosuid,noexec,gid=5,mode=620", 0));
+
+    std::vector<const char*> shellCommandLine{shell.c_str()};
+    std::vector<const char*> env{"TERM=xterm-256color"};
+
+    std::vector<WSLA_PROCESS_FD> fds(3);
+    fds[0].Fd = 0;
+    fds[0].Type = WslFdTypeTerminalInput;
+    fds[1].Fd = 1;
+    fds[1].Type = WslFdTypeTerminalOutput;
+    fds[2].Fd = 2;
+    fds[2].Type = WslFdTypeTerminalControl;
+
+    WSLA_CREATE_PROCESS_OPTIONS processOptions{};
+    processOptions.Executable = shell.c_str();
+    processOptions.CommandLine = shellCommandLine.data();
+    processOptions.CommandLineCount = static_cast<ULONG>(shellCommandLine.size());
+    processOptions.Environment = env.data();
+    processOptions.EnvironmentCount = static_cast<ULONG>(env.size());
+    processOptions.CurrentDirectory = "/";
+
+    std::vector<ULONG> handles(fds.size());
+
+    WSLA_CREATE_PROCESS_RESULT result{};
+    THROW_IF_FAILED(virtualMachine->CreateLinuxProcess(&processOptions, static_cast<ULONG>(fds.size()), fds.data(), handles.data(), &result));
+
+    if (result.Errno != 0)
+    {
+        THROW_HR_WITH_USER_ERROR(E_FAIL, std::format(L"Failed to create process {}, errno = {}", shell.c_str(), result.Errno));
+    }
+
+    // Configure console for interactive usage.
+
+    HANDLE Stdout = GetStdHandle(STD_OUTPUT_HANDLE);
```

Comment:
In theory part of this could be factored with wslrelay's logic, but this is a temporary tool so I'd rather keep this simple